source/core/mesh_controller.cc

```diff
@@ -0,0 +1,47 @@
+#include <core/mesh_controller.h>
+
+
+using namespace std;
+double
+MeshController::calculate_coarsening_factor(int current_number_of_elements)
+{
+  // Parameters of the controller
+  double P = 0.5;
+  double I = 0.3;
+  double D = 0.1;
+
+  // Evaluation of the error used to control the mesh refinement.
+  double error = (target_number_of_elements - current_number_of_elements) /
+                 (double)target_number_of_elements;
```

Comment:
```suggestion
                 (double)target_number_of_elements;
```
Should be a static_cast<double>